op-chain-ops/interopgen/deployers/implementations.go

```diff
@@ -0,0 +1,101 @@
+package deployers
+
+import (
+	"fmt"
+	"math/big"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/script"
+)
+
+type DeployImplementationsInput struct {
+	WithdrawalDelaySeconds          *big.Int
+	MinProposalSizeBytes            *big.Int
+	ChallengePeriodSeconds          *big.Int
+	ProofMaturityDelaySeconds       *big.Int
+	DisputeGameFinalityDelaySeconds *big.Int
+	// Release version to set OPSM implementations for, of the format `op-contracts/vX.Y.Z`.
+	Release               string
+	SuperchainConfigProxy common.Address
+	ProtocolVersionsProxy common.Address
+	UseInterop            bool // if true, deploy Interop implementations
+}
+
+func (input *DeployImplementationsInput) InputSet() bool {
+	return true
+}
+
+type DeployImplementationsOutput struct {
+	Opsm                             common.Address
+	DelayedWETHImpl                  common.Address
+	OptimismPortalImpl               common.Address
+	PreimageOracleSingleton          common.Address
+	MipsSingleton                    common.Address
+	SystemConfigImpl                 common.Address
+	L1CrossDomainMessengerImpl       common.Address
+	L1ERC721BridgeImpl               common.Address
+	L1StandardBridgeImpl             common.Address
+	OptimismMintableERC20FactoryImpl common.Address
+	DisputeGameFactoryImpl           common.Address
+}
+
+func (output *DeployImplementationsOutput) CheckOutput() error {
```

Comment:
Is this meant to be a noop?

Reply:
For now, to just get it running, yes. But we can add a sanity-check that all the fields are set etc. Can also do checks after script execution, instead of this call that happens during the script.